generic3g/specs/VariableSpec.F90

```diff
@@ -199,11 +203,14 @@ function make_ItemSpec_new(this, geom, vertical_grid, registry, rc) result(item_
 
       integer :: status
       type(ActualPtVector) :: dependencies
+      type(ESMF_Geom), allocatable :: geom_local
+
+      call this%pick_geom_(geom, geom_local, _RC)
```

Comment:
One subtle problem.   If `pick_geom` fails, then the allocatable function result (`item_spec`) is not allocated.  This creates errors upstream that cannot be trapped.

The procedure now needs a temporary assignment of `item_spec = NullSpec()` except ... that triggers a known problem with gfortran and reallocation of poly items that means you have to deallocate it again before each of the allocations in the SELECT CASE construct.  Ugh.

I'll approve but please open a ticket to fix this upon your return.

Reply:
Created the issue.

I have been struggling with these Fortran subtleties all of last week :-(. My understanding of Fortran is not what I thought it was.

Reply:
> I have been struggling with these Fortran subtleties all of last week :-(. My understanding of Fortran is not what I thought it was.

It is not that your understanding was inadequate, but rather that the language (and my use of it) has opened new ways for things to fail.   Functions that can fail are really a bad idea.  But I like functional programming, and keep trying to apply it in a software ecosystem that is saturated with failure conditions.